pkgs/development/tools/build-managers/meson/default.nix

```diff
@@ -63,7 +65,33 @@ python3.pkgs.buildPythonApplication rec {
     ./do-not-update-ldconfig-cache.patch
   ];
 
-  setupHook = ./setup-hook.sh;
+  cpuFamily = with stdenv.targetPlatform;
+    /**/ if isAarch32 then "arm"
+    else if isAarch64 then "aarch64"
+    else if isx86_32  then "x86"
+    else if isx86_64  then "x86_64"
+    else parsed.cpu.family + builtins.toString parsed.cpu.bits;
+
+  crossFile = if stdenv.hostPlatform == stdenv.targetPlatform then null else
```

Comment:
Cannot really comment much about this other than that the cross file was in Meson expression [in the past](https://github.com/NixOS/nixpkgs/commit/8245230753d260b61b928fad7e00323f38ca450f). Not sure, what the reasons were for the move. cc @Ericson2314

Reply:
Yeah putting it in `stdenv` is gross, but i think it is better so that we avoid a target-sensative meson.

Reply:
@matthewbauer and I did this, and opened up an issue for "nix setup hooks", so that we could have some modularity without trying to do this stuff in bash instead of Nix